wire-ios-request-strategy/Sources/Request Strategies/Conversation/ConversationEventProcessor.swift

```diff
@@ -59,116 +60,168 @@ public class ConversationEventProcessor: NSObject, ConversationEventProcessorPro
-                    await context.perform {
-                        self.processor.processPayload(
-                            payload,
-                            in: self.context
-                        )
-                    }
-                }
-
-            case .conversationConnectRequest:
-                if let payload = event.eventPayload(type: Payload.ConversationEvent<Payload.UpdateConversationConnectionRequest>.self) {
-                    await context.perform {
-                        self.processor.processPayload(
-                            payload,
-                            originalEvent: event,
-                            in: self.context
-                        )
-                    }
-                }
-
-            case .conversationMLSWelcome:
-                guard
-                    let data = event.payloadData,
-                    let payload = Payload.UpdateConversationMLSWelcome(data)
-                else {
-                    break
-                }
-
-                await MLSEventProcessor.shared.process(
-                    welcomeMessage: payload.data,
-                    in: self.context
-                )
+            await processConversationEvent(event)
+        }
+    }
 
-            default:
-                break
-            }
+    private func processConversationEvent(_ event: ZMUpdateEvent) async {
+        switch event.type {
+        case .conversationCreate:
+            await processConversationCreate(event)
+
+        case .conversationDelete:
+            await processConversationDelete(event)
+
+        case .conversationMemberLeave:
+            await processConversationMemberLeave(event)
+
+        case .conversationMemberJoin:
+            await processConversationMemberJoin(event)
+
+        case .conversationRename:
+            await processConversationRename(event)
+
+        case .conversationMemberUpdate:
+            await processConversationMemberUpdate(event)
+
+        case .conversationAccessModeUpdate:
+            await processConversationAccessModeUpdate(event)
+
+        case .conversationMessageTimerUpdate:
+            await processConversationMessageTimerUpdate(event)
+
+        case .conversationReceiptModeUpdate:
+            await processConversationReceiptModeUpdate(event)
+
+        case .conversationConnectRequest:
+            await processConversationConnectRequest(event)
+
+        case .conversationMLSWelcome:
+            await processConversationMLSWelcome(event)
+
+        default:
+            break
+        }
+    }
+
+    private func processConversationCreate(_ event: ZMUpdateEvent) async {
+        guard let payload = try? eventPayloadDecoder.decode(Payload.ConversationEvent<Payload.Conversation>.self, from: event.payload) else {
+            return
+        }
+        await processor.processPayload(payload, in: context)
```

Comment:
What do you think about having this on multiple lines to make it more readable?
```suggestion
        guard let payload = try? eventPayloadDecoder.decode(
            Payload.ConversationEvent<Payload.Conversation>.self, 
            from: event.payload
        ) else { return }
        await processor.processPayload(payload, in: context)
```

Reply:
Thanks, I updated the formatting